exonum/src/blockchain/transaction.rs

```diff
@@ -37,7 +37,8 @@ const TRANSACTION_STATUS_PANIC: u16 = TRANSACTION_STATUS_OK + 1;
 pub type ExecutionResult = Result<(), ExecutionError>;
 /// Extended version of `ExecutionResult` (with additional values set exclusively by Exonum
 /// framework) that can be obtained through `Schema` `transaction_statuses` method.
-pub type TransactionResult = Result<(), TransactionError>;
+#[derive(Clone, Debug, PartialEq)]
+pub struct TransactionResult(pub Result<(), TransactionError>);
```

Comment:
I'm not happy to use newtype struct in this place, but we can't use `TransactionResult` in way we used it before because of need to `impl CryptoHash` on it.
TODO: think about how to improve it.

Reply:
It's ok that `TransactionResult` is now a new type. 
But can you provide `Into<Result<(), TransactionError>> for TransactionResult ` `Asref` and the rest helper traits that could safe us from manual wrapping and unwrapping results?

Reply:
Ok, no one of this trait can be useful for us because of `Cannot move out of borrowed content` for `Into` and `AsRef`, so I'm leaving it as is.

Reply:
To make it really transparent we could use `Deref` into `Result`, but TBH that would be abuse of the trait originally meant for smart pointers. But I don't really see why we can't add

```rust
impl From<Result<(), TransactionError>> for TransactionResult {
    fn from(error: Result<(), TransactionError>) -> TransactionResult {
        TransactionResult(error)
    }
}
```

to make `?` and `into()` work.